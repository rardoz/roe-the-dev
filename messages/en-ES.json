{
  "Home": {
    "title": "Roe the Dev. Hola!"
  }
}
