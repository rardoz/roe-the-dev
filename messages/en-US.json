{
  "Home": {
    "title": "Roe the Dev"
  },
  "Nav": {
    "portfolio": "Porfolio",
    "blog": "Blog",
    "experiments": "Experiments",
    "menuIconAltText": "Click for additional menu links",
    "logoAltText": "Roe the Dev logo",
    "about": "About Roe"
  },
  "About": {
    "title": "About Roe",
    "subtitle": "Tech Lead, Software Engineer, Mentor And Artist",
    "description1": "I am a versatile digital nomad and innovative creative technologist assuming the role of a tech lead while embracing a nomadic lifestyle, dedicated to bringing creative ideas to fruition.",
    "description2": "I pursued my education at Black Hawk College and The Illinois Institute of Art in Chicago, achieving an associate's degree in visual communications. With a professional tenure of over a decade, I have accumulated substantial experience in software development.",
    "cta": "Learn more"
  },
  "Portfolio": {
    "title": "Portfolio",
    "cta": "See my work",
    "footerMessage": "Background AI generated with"
  },
  "Blog": {
    "title": "From my desk",
    "cta": "Read blogs",
    "footerMessage": "Background AI generated with"
  }
}
