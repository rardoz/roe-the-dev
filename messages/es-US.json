{
  "Home": {
    "title": "Roe the Dev. Hola!"
  },
  "Nav": {
    "portfolio": "Cartera",
    "blog": "Blog",
    "experiments": "Experimentación",
    "menuIconAltText": "Haga clic para ver más enlaces",
    "logoAltText": "Logotipo de Roe the Dev",
    "about": "Sobre Roe"
  },
  "About": {
    "title": "Sobre Roe",
    "subtitle": "Líder Tecnológico, Ingeniero de Software, Mentor y Artista",
    "description1": "Soy un nómada digital versátil e innovador tecnólogo creativo, asumiendo el rol de líder tecnológico mientras abrazo un estilo de vida nómada, dedicado a llevar a cabo ideas creativas.",
    "description2": "Cursé mis estudios en Black Hawk College y en The Illinois Institute of Art en Chicago, obteniendo un título de asociado en comunicación visual. Con una experiencia profesional de más de una década, he acumulado una considerable experiencia en desarrollo de software.",
    "cta": "Aprender más"
  },
  "Portfolio": {
    "title": "Portafolio",
    "cta": "Ver mi trabajo",
    "footerMessage": "Fondo generado por inteligencia artificial con"
  },
  "Blog": {
    "title": "Desde mi escritorio",
    "cta": "Leer blogs"
  }
}
