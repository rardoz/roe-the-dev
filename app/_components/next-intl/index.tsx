import { createLocalizedPathnamesNavigation } from 'next-intl/navigation'

import config from '../../../messages/config'

export const { Link, redirect, usePathname, useRouter, getPathname } =
  createLocalizedPathnamesNavigation({
    locales: config.locales,
    pathnames: config.pathnames,
  })

import { ComponentProps } from 'react'
import type { AppPathnames } from '../../../messages/config'

export function UnstyledLink<Pathname extends AppPathnames>({
  href,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return <Link href={href} {...rest} />
}

export function StyledLink<Pathname extends AppPathnames>({
  className,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return (
    <UnstyledLink
      {...rest}
      className={`transition-colors duration-500 text-pink-500 hover:text-pink-400 ${
        className || ''
      }`}
    />
  )
}
