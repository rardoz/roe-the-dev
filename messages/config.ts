import { Pathnames } from 'next-intl/navigation'

const localePrefix: 'as-needed' | 'always' | 'never' | undefined = 'as-needed'

export const locales = ['en-US', 'es-US']

export type AppPathname = Pathnames<typeof locales>

export const pathnames = {
  '/blog': '/blog',
  '/blog/[slug]': '/blog/[slug]',
  '/portfolio': '/portfolio',
  '/portfolio/[slug]': '/portfolio/[slug]',
  '/experiment': '/experiment',
  '/': { 'en-US': '/', 'es-US': '/' },
} satisfies AppPathname

export type AppPathnames = keyof typeof pathnames

const config = {
  locales,
  defaultLocale: 'en-US',
  localePrefix,
  pathnames,
}

export default config
