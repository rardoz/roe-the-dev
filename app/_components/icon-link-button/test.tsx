import IconLinkButton from '.'
import { render } from '@testing-library/react'
import { NextIntlClientProvider } from 'next-intl'
import messages from '../../../messages/en-US.json'

describe('IconLinkButton', () => {
  it('should render with default locale set', () => {
    expect(
      render(
        <NextIntlClientProvider locale="en-US" messages={messages}>
          <IconLinkButton href="/blog" altText="blog">
            Blog
          </IconLinkButton>
        </NextIntlClientProvider>,
      ).container,
    ).toMatchSnapshot()
  })

  it('should include a className if set', () => {
    expect(
      render(
        <NextIntlClientProvider locale="en-US" messages={messages}>
          <IconLinkButton href="/blog" altText="blog" className="test">
            Blog
          </IconLinkButton>
        </NextIntlClientProvider>,
      ).container,
    ).toMatchSnapshot()
  })
})
