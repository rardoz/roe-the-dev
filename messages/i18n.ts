import { getRequestConfig } from 'next-intl/server'
import { default as config, locales } from './config'

// export default getRequestConfig(async ({ requestLocale }) => ({
//   messages: (await import(`./${requestLocale}.json`)).default,
// }))

export default getRequestConfig(async ({ requestLocale }) => {
  let _locale = await requestLocale
  if (!_locale || locales.includes(_locale as any)) {
    _locale = config.defaultLocale
  }

  return {
    locale: _locale,
    messages: (await import(`./${_locale}.json`)).default,
  }
})
