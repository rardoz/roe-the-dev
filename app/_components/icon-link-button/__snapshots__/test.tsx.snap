// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLinkButton should include a className if set 1`] = `
<div>
  <a
    class="w-12 h-12 justify-center mx-1 my-1 flex items-center transition-colors duration-500 hover:bg-pink-600 text-white bg-purple-700 rounded-full dark:bg-blue-600 test"
    href="/en-US/blog"
  >
    <div
      class="sr-only"
    >
      blog
    </div>
    Blog
  </a>
</div>
`;

exports[`IconLinkButton should render with default locale set 1`] = `
<div>
  <a
    class="w-12 h-12 justify-center mx-1 my-1 flex items-center transition-colors duration-500 hover:bg-pink-600 text-white bg-purple-700 rounded-full dark:bg-blue-600 "
    href="/en-US/blog"
  >
    <div
      class="sr-only"
    >
      blog
    </div>
    Blog
  </a>
</div>
`;
